aea/cli/generate.py

```diff
@@ -131,30 +126,27 @@ def _generate_item(click_context, item_type, specification_path):
             ctx.agent_config, open(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w")
         )
     except FileExistsError:
-        logger.error(
+        raise click.ClickException(
             "A {} with this name already exists. Please choose a different name and try again.".format(
                 item_type
             )
         )
-        sys.exit(1)
     except ProtocolSpecificationParseError as e:
-        logger.error(
+        raise click.ClickException(
             "The following error happened while parsing the protocol specification: "
             + str(e)
         )
         shutil.rmtree(
             os.path.join(item_type_plural, protocol_spec.name), ignore_errors=True
         )
-        sys.exit(1)
     except Exception as e:
         logger.debug("Exception thrown: " + str(e))
-        logger.error(
-            "There was an error while generating the protocol. The protocol is NOT generated."
-        )
         shutil.rmtree(
             os.path.join(item_type_plural, protocol_spec.name), ignore_errors=True
         )
-        sys.exit(1)
+        raise click.ClickException(
```

Comment:
Remove debug and print exception message here

Reply:
Sorry I think you misunderstood, the exception message should be moved inside the click exception, i.e.: `"There was an error while generating the protocol. The protocol is NOT generated. Exception thrown: " + str(e)`

Reply:
Oh yes, now I see. Will do.